File: ILT.IHR.DTO/Asset.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace ILT.IHR.DTO
{
    public partial class Asset : AbstractDataObject
    {
        public int AssetID
        {
            get { return base.RecordID; }
            set { base.RecordID = value; }
        }
        [Required]
        [RegularExpression("^0*[1-9]\\d*$")]
        public int AssetTypeID { get; set; }
        public string AssetType { get; set; }
        [Required]
        public string Tag { get; set; }
        public string Make { get; set; }
        public string Model { get; set; }
        [Required]
        public string Configuration { get; set; }
        public string WiFiMAC { get; set; }
        public string LANMAC { get; set; }
        public string OS { get; set; }
        [Required]
        public DateTime PurchaseDate { get; set; }
        public DateTime? WarantyExpDate { get; set; }
       // [RegularExpression("^0*[1-9]\\d*$")]
        public int? AssignedToID { get; set; }
        public string AssignedTo { get; set; }
        [Required]
        [RegularExpression("^0*[1-9]\\d*$")]
        public int StatusID { get; set; }
        public string Status { get; set; }
        public string Comment { get; set; }
    }
}

File: ILT.IHR.APP/ClientApp/src/app/core/interfaces/Asset.ts
export class IAsset {
    AssetID: number;
    AssetTypeID: number;
    AssetType: string;
    Tag: string;
    Make?: string;
    Model?: string;
    Configuration: string;
    WiFiMAC: string;
    LANMAC: string;
    OS: string;
    PurchaseDate: Date;
    WarantyExpDate: Date;
    AssignedToID: number;
    AssignedTo: string;
    StatusID: number;
    Status: string;
    Comment: string;
    CreatedBy: string;
    CreatedDate: Date;
    ModifiedBy: string;
    ModifiedDate: Date;
    TimeStamp: ['']

}
export class IAssetDisplay {
    assetID: number;
    assetTypeID: number;
    assetType: string;
    tag: string;
    make?: string;
    model?: string;
    configuration: string;
    wiFiMAC: string;
    lanmac: string;
    os: string;
    purchaseDate: Date;
    warantyExpDate: Date;
    assignedToID: number;
    assignedTo: string;
    statusID: number;
    status: string;
    comment: string;
    createdBy: string;
    createdDate: Date;
    modifiedBy: string;
    modifiedDate: Date;
    timeStamp: ['']
}
